feature-libs/customer-ticketing/components/details/customer-ticketing-details/customer-ticketing-details.component.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * SPDX-FileCopyrightText: 2022 SAP Spartacus team <spartacus-team@sap.com>
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ChangeDetectionStrategy, Component, OnDestroy } from '@angular/core';
+import {
+  EventService,
+  RoutingService,
+  TranslationService,
+} from '@spartacus/core';
+import {
+  CustomerTicketingFacade,
+  DATE_FORMAT,
+  GetTicketQueryReloadEvent,
+  STATUS,
+  TEXT_COLOR_CLASS,
+  TicketDetails,
+} from '@spartacus/customer-ticketing/root';
+import { Card } from '@spartacus/storefront';
+import { combineLatest, Observable, Subscription } from 'rxjs';
+import { filter, map, take, tap } from 'rxjs/operators';
+
+@Component({
+  selector: 'cx-customer-ticketing-details',
+  templateUrl: './customer-ticketing-details.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CustomerTicketingDetailsComponent implements OnDestroy {
+  dateFormat = DATE_FORMAT;
+  subscription = new Subscription();
+  ticketDetails$: Observable<TicketDetails | undefined> =
+    this.customerTicketingFacade.getTicket();
+
+  constructor(
+    protected translation: TranslationService,
+    protected customerTicketingFacade: CustomerTicketingFacade,
+    protected routingService: RoutingService,
+    protected eventService: EventService
+  ) {
+    this.reloadOnRedirection();
+  }
+
+  prepareCardContent(
+    entity: string,
+    titleTranslation: string,
+    id?: string
+  ): Observable<Card> {
+    return this.translation.translate(titleTranslation).pipe(
+      filter(() => Boolean(entity)),
+      map((textTitle) => ({
+        title: textTitle,
+        text: [entity],
+        customClass: this.getStatusClass(id?.toUpperCase()),
+      }))
+    );
+  }
+
+  getStatusClass(id?: string): string {
+    if (id === STATUS.OPEN || id === STATUS.INPROCESS) {
+      return TEXT_COLOR_CLASS.GREEN;
+    } else if (id === STATUS.CLOSED) {
+      return TEXT_COLOR_CLASS.GREY;
+    }
+    return '';
+  }
+
+  protected reloadOnRedirection() {
```

Comment:
good initiative. Will think about it because it's already too late to reload as you'll see the old state for a few seconds before it being updated due to the component initialization happens at the time it hits the page with the component, and not before.

Reply:
This implementation came in to picture because when user opens a ticket, go back to ticket list and opens another ticket, the data was not reloading.